apps/timetrace/ttanal

```diff
@@ -32,6 +33,46 @@ def tt_reader(paths: list[str]):
                     break
                 yield buf_to_rec(buf)
 
+time_units = [
+    ("ns", 0.000000001),
+    ("us", 0.000001),
+    ("ms", 0.001),
+    ("s", 1),
+    ("m", 60),
+    ("h", 3600),
+    ("d", 86400),
+    ("y", 31556952),
+    (None, None),
+]
+
+mem_units = [
+    ("B", 1),
+    ("kB", 1024),
+    ("MB", 1048576),
+    ("GB", 1073741824),
+    ("TB", 1099511627776),
+    (None, None),
+]
+
+def pretty_num(units: list[tuple[str, float]], s: float) -> str:
+    if s == 0:
+        return "0"
+    sign = ""
+    if s < 0:
+        sign = "-"
+        s *= -1
+    for i, (unit, val) in enumerate(units):
+        nxtval = units[i + 1][1]
+        if nxtval is None or s < nxtval:
+            s /= val
+            return f"{sign}{s:.3g}{unit}"
+
+def pretty_time(s: float) -> str:
+    return pretty_num(time_units, s)
+
+def pretty_mem(s: float) -> str:
+    return pretty_num(mem_units, s)
+
 class Foo:
```

Comment:
I must be missing something here but why is this a class in the first place?

Reply:
It's a habit, because globals are bad and encapsulation is good.  I forgot to take advantage.  Update coming momentarily.